src/bin/util.c

```diff
@@ -81,38 +82,84 @@ static const char *basenm(const char *fn)
 	return ptr;
 }
 
-char *cfg_adjust(const char *fn, const char *tmpl, char *buf, size_t len)
+char *cfg_adjust(const char *fn, const char *tmpl, char *buf, size_t len, int sanitize)
 {
-	if (strstr(fn, "../"))
-		return NULL;	/* relative paths not allowed */
-
-	if (fn[0] == '/') {
-		strlcpy(buf, fn, len);
-		return buf;	/* allow absolute paths */
-	}
-
-	/* Files in /cfg must end in .cfg */
-	if (!strncmp(fn, "/cfg/", 5)) {
+	if (sanitize) {
+		char testpath[PATH_MAX];
+		/* CLI mode - restricted and helpful */
+
+		/* Block path traversal attacks */
+		if (strstr(fn, "../"))
```

Comment:
Just a heads up. If this path is later used together with something, like `("%s/%s", fn, name)` an attacker could place `..` in `fn` like `/etc/sysrepo/..` and then set `name` to `shadow`. This would bypass this check and result in `/etc/sysrepo/../shadow`, or if he/she can control the name even `(%s%s", fn, name)` would be vulnerable as name could be `/shadow`.

Not that I believe it's a problem here.